qtaskbarcontrol.cpp

```diff
@@ -5,14 +5,17 @@
 #include <QWidget>
 #include <QIcon>
 
-QTaskbarControl::QTaskbarControl(QWidget *parent) :
+QTaskbarControl::QTaskbarControl(QObject *parent) :
 	QObject{parent},
-	d{QTaskbarControlPrivate::createPrivate(this)}
+	d{QTaskbarControlPrivate::createPrivate()}
+{}
+
+void QTaskbarControl::setWindow(QWidget *widget)
 {
-	Q_ASSERT_X(parent, Q_FUNC_INFO, "QTaskbarControl must have a valid parent");
-	parent->installEventFilter(this);
-	if(parent->windowHandle())
-		d->setWindow(parent->windowHandle());
+	if (widget->windowHandle())
+		d->setWindow(widget->windowHandle());
+	else
+		widget->installEventFilter(this);
```

Comment:
Please undo these lines. As already explained in the comment in line 119, a widgets window handle might change, if it gets reparented. Always using the event filter makes sure, the taskbar control gets notified of this change.

Reply:
Also, we need to remove the eventfilter if the widget changes, as this method could be called multiple times. I think adding a private member that holds a `QPointer<QWidget>` and gets set and unset in this method.

Reply:
I agree with you. I will add it.